core/src/main/scala/com/pennsieve/aws/s3/S3.scala

```diff
@@ -301,7 +305,24 @@ class S3(val client: AmazonS3) extends S3Trait {
           .withRequesterPays(isRequesterPays)
         client.deleteObjects(request)
       }
+      deleteAllVersions(bucket,key,client)
+    }
+
+  // Gathers up all versions of a $key and does a hard delete
+def deleteAllVersions(bucket: String, key: String, client: AmazonS3): Either[Throwable, Unit] = {
+  try {
+    val versionListing: VersionListing = client.listVersions(bucket, key)
+    val deleteRequests = versionListing.getVersionSummaries.asScala.map { versionSummary =>
+      new DeleteVersionRequest(bucket, key, versionSummary.getVersionId)
     }
+    deleteRequests.foreach(request => client.deleteVersion(request))
+    println(s"Permanently deleted all versions of object $key.")
```

Comment:
can this be made a log statement instead of println? (not sure if a logger implicit is available here, might need to be done higher in the call stack)

Reply:
I removed the println and added a log higher up in the call stack